datajunction-query/djqs/config.py

```diff
@@ -3,94 +3,202 @@
 
-    # The default engine to use for reflection
-    default_reflection_engine: str = "default"
+class EngineInfo:  # pylint: disable=too-few-public-methods
+    """
+    Information about a query engine
+    """
 
-    # The default engine version to use for reflection
-    default_reflection_engine_version: str = ""
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        name: str,
+        version: str,
+        type: str,  # pylint: disable=redefined-builtin
+        uri: str,
+        extra_params: Optional[Dict[str, str]] = None,
+    ):
+        self.name = name
+        self.version = str(version)
+        self.type = EngineType(type)
+        self.uri = uri
+        self.extra_params = extra_params or {}
+
+
+class CatalogInfo:  # pylint: disable=too-few-public-methods
+    """
+    Information about a catalog
+    """
 
-    # Where to store the results from queries.
-    results_backend: BaseCache = FileSystemCache("/tmp/djqs", default_timeout=0)
+    def __init__(self, name: str, engines: List[str]):
+        self.name = name
+        self.engines = engines
 
-    paginating_timeout: timedelta = timedelta(minutes=5)
 
-    # How long to wait when pinging databases to find out the fastest online database.
-    do_ping_timeout: timedelta = timedelta(seconds=5)
+class Settings:  # pylint: disable=too-many-instance-attributes
+    """
+    Configuration for the query service
+    """
 
-    # Configuration file for catalogs and engines
-    configuration_file: Optional[str] = None
+    def __init__(  # pylint: disable=too-many-arguments,too-many-locals,dangerous-default-value
+        self,
+        name: Optional[str] = "DJQS",
+        description: Optional[str] = "A DataJunction Query Service",
+        url: Optional[str] = "http://localhost:8001/",
+        index: Optional[str] = "postgresql://dj:dj@postgres_metadata:5432/dj",
+        alembic_uri: Optional[
+            str
+        ] = "postgresql+psycopg://dj:dj@postgres_metadata:5432/dj",
+        default_catalog: Optional[str] = "",
+        default_engine: Optional[str] = "",
+        default_engine_version: Optional[str] = "",
+        results_backend: Optional[BaseCache] = None,
+        results_backend_path: Optional[str] = "/tmp/djqs",
+        results_backend_timeout: Optional[str] = "0",
+        paginating_timeout_minutes: Optional[str] = "5",
+        do_ping_timeout_seconds: Optional[str] = "5",
+        configuration_file: Optional[str] = None,
+        engines: Optional[List[EngineInfo]] = None,
+        catalogs: Optional[List[CatalogInfo]] = None,
+    ):
+        self.name: str = os.getenv("NAME", name or "")
+        self.description: str = os.getenv("DESCRIPTION", description or "")
+        self.url: str = os.getenv("URL", url or "")
+
+        # SQLAlchemy URI for the metadata database.
+        self.index: str = os.getenv("INDEX", index or "")
+
+        self.alembic_uri: str = os.getenv("ALEMBIC_URI", alembic_uri or "")
```

Comment:
It seems like this could default to self.index if it's not set here?

Reply:
Fixed by removing alembic_uri, it's not actually needed anymore.